core/trino-spi/src/main/java/io/trino/spi/predicate/SortedRangeSet.java

```diff
@@ -842,11 +842,26 @@ public String toString(ConnectorSession session)
 
     private String formatRanges(ConnectorSession session, int limit)
     {
-        return Stream.concat(
-                IntStream.range(0, min(getRangeCount(), limit))
+        if (isNone()) {
+            return "{}";
+        }
+        if (getRangeCount() == 1) {
+            return getRangeView(0).formatRange(session);
+        }
+        if (limit < 2) {
+            return "span=" + getSpan().toString(session);
+        }
+        // Print first (limit - 1) elements, followed by last element
```

Comment:
Neat. Why? Document.

Reply:
added a comment

Reply:
I agree that the last range is more interesting than say 10th.